.github/workflows/gradle.yml

```diff
@@ -0,0 +1,32 @@
+# CI workflow
+
+name: Build
+
+# Controls when the action will run. 
+on:
+  # Triggers the workflow on push or pull request events but only for the master branch
+  push:
+    branches: [ master ]
```

Comment:
I think it's also useful to detect compile errors in branches with pull requests to fix them before validation : Example on adding action to branches : 
        # Push events to branches matching refs/heads/releases/10
              - 'releases/**'

https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#onpushpull_requestbranchestags